app/models/filter.rb

```diff
@@ -50,7 +50,7 @@ def ensure_taxonomies_not_escalated
   scoped_search :relation => :permissions, :on => :name,          :rename => :permission
 
   before_validation :build_taxonomy_search, :nilify_empty_searches, :enforce_override_flag
-  before_save :enforce_inherited_taxonomies
+  before_save :enforce_inherited_taxonomies, :nilify_empty_searches
```

Comment:
Rearranging these filters makes other tests in filter_test.rb fail - so opted to just duplicate the nilify here. It's better than my first fix since it's less specific

Reply:
:+1: cool